pymc/model.py

```diff
@@ -1409,67 +1407,27 @@ def make_obs_var(
             if total_size is not None:
                 raise ValueError("total_size is not compatible with imputed variables")
 
-            if not isinstance(rv_var.owner.op, RandomVariable):
-                raise NotImplementedError(
-                    "Automatic inputation is only supported for univariate RandomVariables."
-                    f" {rv_var} of type {type(rv_var.owner.op)} is not supported."
-                )
-
-            if rv_var.owner.op.ndim_supp > 0:
-                raise NotImplementedError(
-                    f"Automatic inputation is only supported for univariate "
-                    f"RandomVariables, but {rv_var} is multivariate"
-                )
+            from pymc.distributions.distribution import create_partial_observed_rv
 
-            # We can get a random variable comprised of only the unobserved
-            # entries by lifting the indices through the `RandomVariable` `Op`.
+            (
+                (observed_rv, observed_mask),
+                (unobserved_rv, _),
+                joined_rv,
+            ) = create_partial_observed_rv(rv_var, mask)
+            observed_data = pt.as_tensor(data.data[observed_mask])
 
-            masked_rv_var = rv_var[mask.nonzero()]
-
-            fgraph = FunctionGraph(
-                [i for i in graph_inputs((masked_rv_var,)) if not isinstance(i, Constant)],
-                [masked_rv_var],
-                clone=False,
-            )
+            # Register ObservedRV corresponding to observed component
+            observed_rv.name = f"{name}_observed"
+            self.create_value_var(observed_rv, transform=None, value_var=observed_data)
+            self.add_named_variable(observed_rv)
+            self.observed_RVs.append(observed_rv)
 
-            (missing_rv_var,) = local_subtensor_rv_lift.transform(fgraph, fgraph.outputs[0].owner)
+            # Register FreeRV corresponding to unobserved components
+            self.register_rv(unobserved_rv, f"{name}_missing", transform=transform)
```

Comment:
```suggestion
            self.register_rv(unobserved_rv, f"{name}_unobserved", transform=transform)
```
this would match the `_observed` for the other and maybe avoid confusion?

Reply:
This could count as a breaking change, that's why I didn't change it. Worth the trouble?

Reply:
I do think it's a better name

Reply:
I'd say it's okay to break. Imputation isn't that popular, and the breaking change isn't dangerously silent or sth like that.

Reply:
It changes the variable name, that's all

Reply:
Updated